omero/developers/Web/Webclient.txt

```diff
@@ -0,0 +1,90 @@
+OMERO.webclient
+===============
+
+The webclient Django app provides the main OMERO.web client UI.
+
+Data from OMERO is rendered to html using Django templates before being
+sent to the browser. Additional javascript (jQuery-based) functionality in the browser
+is used to update the UI in response to user interactions such as 
+browsing or modifying the data.
+
+.. note::
+    The webclient should **NOT** be considered as a stable public API. URLs and methods
+    within the app are purely designed to provide internal functionality for
+    the webclient itself and may change in minor releases without warning.
+
+Top level pages
+---------------
+
+There are a small number of top level html pages that the user will start at.
+These are all handled by the load_template view:
+
+- / (homepage delegates to /userdata)
+- /userdata
+- /usertags
+- /public
+- /history
+- /search
+
+These pages contain many different jQuery scripts that run when the page loads,
+to setup event listeners and to load additional data.
+
+Additional top-level pages are used in popup windows for running scripts and
+downloading data.
+
+JsTree
+------
+
+A jsTree is used by the userdata, usertags and public pages to browse hierarchical
+data. Each time a node is expanded, it uses appropriate AJAX calls to load children as
+json data.
+Further POST or DELETE AJAX calls are made to modify the data hierachy by
+creating or deleting links between objects.
+
+Selection changes in the jsTree and centre panel thumbnails cause events to be
+triggered by jQuery on the ``$("body")`` element of the page, allowing other scripts
+to listen for these events.
+These are used to load selected data into the centre and right panels.
+The html for these panels contains additional scripts that also run when they load
+to setup their own event listeners.
+
+There is also a global ``update_thumbnails_panel`` function that can be called
+by any script that needs to refresh the centre panel. For example, when the jsTree
+is used to add or remove Images from a selected Dataset.
+
+Switching Groups and Users
+--------------------------
+
+The current group and user are stored in the html session as ``active_group`` and
```

Comment:
The current group and user are stored in the :djangodoc:`HTTP session <topics/http/sessions/>` and maybe give examples ``request.session['active_group']``, ``request.session['user_id']``